### ISSUE TITLE
Precision Issues with GPTQ-Quantized Qwen2.5-VL Model

### ISSUE THREAD

**Issue by KarlDe1** [vllm-project/llm-compressor#1629]
After applying GPTQ quantization to the Qwen2.5-VL model and running it on the test set, I observed a drop in accuracy. Are there any methods to help identify or locate the source of this accuracy degradation?

In this situation, should we be trying to **locate where exactly the accuracy issue is introduced**, or is this kind of degradation **expected/normal** after quantization?

**Environment**
Include all relevant environment information:
1. Ubuntu22.04
2. Python version:3.10.16
3. LLM Compressor version: v0.6.0
4. torch version: 2.6.0+cu124
5. vLLM version: 0.8.5
6. CUDA version: 12.8
7. transformers version: 4.52.4
8. GPU: nvidia H20

```
recipe = [
    GPTQModifier(
        targets="Linear",
        scheme="W4A16",
        sequential_targets=["Qwen2_5_VLDecoderLayer"],
        ignore=["lm_head", "re:visual.*"],
    ),
]
```

**Comment by KarlDe1** [comment]
My prompt provides an image and asks the model to make a driving decision. After testing on the dataset, I found that in **3% of the cases**, **the quantized model made incorrect decisions**, whereas **the original (non-quantized) model made the correct choices**.

**Comment by kylesayrs** [comment]
Hi @KarlDe1 

You can see the RedhatAI official models uploaded [here](https://huggingface.co/RedHatAI/models?search=qwen2.5-vl). Each of these model cards shows the measured accuracy as well as performance expectations alongside each scheme.

Hopefully this helps!

**Comment by KarlDe1** [comment]
Hi @kylesayrs 

Thank you for your reply. Is it possible to recover some of the accuracy through certain methods? For example, by ignoring **more** layers during quantization?

**Comment by KarlDe1** [comment]
Hi @kylesayrs 

Also, when I was quantizing the model using the script provided [here](https://huggingface.co/RedHatAI/models?search=qwen2.5-vl) by llm-compressor-v0.6.0, I noticed the following log. I believe the intention was to exclude **the visual-related layers** from quantization, so perhaps the pattern should be updated to **"re:model.visual.*"**?

<img width="3082" height="1164" alt="Image" src="https://github.com/user-attachments/assets/c800fc34-add2-4863-acb6-7c23c3912ddf" />

<img width="1432" height="1368" alt="Image" src="https://github.com/user-attachments/assets/c48dce53-ea0e-40c5-a943-b63f2432894a" />

**Comment by KarlDe1** [comment]
I tested the **ChartQA (test, CoT)** dataset using a quantized model - Qwen2.5-VL-3B-Instruct-W4A16, and obtained an **anywhere_in_answer_relaxed_correctness** score of **71.96**, which is lower than the **78.96** reported in RedHatAI's paper. 
```bash
================================================================================
Metrics:
{
    "explicit_prompt_relaxed_correctness": 0.7004,
    "anywhere_in_answer_relaxed_correctness": 0.7196
}
================================================================================
```
However, the score from the original (non-quantized) model matches the one reported in the RedHatAI paper.

<img width="1898" height="1580" alt="Image" src="https://github.com/user-attachments/assets/58c64a93-df99-4175-b8e5-157471e3b0e8" />

I tested Qwen2.5-VL-3B-Instruct-W4A16 using **MMMU (val, CoT)**, and the results are basically consistent with those reported in the paper. However, during the evaluation, there were many **400 Bad Request** errors. I'm not sure if you also encountered this issue during your testing.
```bash
================================================================================
Metrics:
{
    "explicit_prompt_relaxed_correctness": 0.431924882629108,
    "anywhere_in_answer_relaxed_correctness": 0.4518779342723005
}
================================================================================
```

The quantization code used in the RedHatAI report is no longer compatible with the latest versions of llm-compressor and vLLM, so I used the updated quantization code for quantization.

**Environment**
llm-compressor v0.6.0
vLLM v0.9.2rc1
<details>

<summary>Quantization Code</summary>

```python
import base64
from io import BytesIO

import torch
from datasets import load_dataset
from qwen_vl_utils import process_vision_info
from transformers import AutoProcessor, Qwen2_5_VLForConditionalGeneration

from llmcompressor.modifiers.quantization import GPTQModifier
from llmcompressor.transformers import oneshot
from llmcompressor.utils import dispatch_for_generation

# Load model.
model_id = "/vlm/model/Qwen2.5-VL-3B-Instruct"
model = Qwen2_5_VLForConditionalGeneration.from_pretrained(model_id, torch_dtype="auto")
processor = AutoProcessor.from_pretrained(model_id, trust_remote_code=True)

# Oneshot arguments
DATASET_ID = "/vlm/dataset/flickr30k"
DATASET_SPLIT = "test[:512]"
NUM_CALIBRATION_SAMPLES = 512
MAX_SEQUENCE_LENGTH = 4096

# Load dataset and preprocess.
ds = load_dataset(DATASET_ID, split=DATASET_SPLIT)
ds = ds.shuffle(seed=42)


# Apply chat template and tokenize inputs.
def preprocess_and_tokenize(example):
    # preprocess
    buffered = BytesIO()
    example["image"].save(buffered, format="PNG")
    encoded_image = base64.b64encode(buffered.getvalue())
    encoded_image_text = encoded_image.decode("utf-8")
    base64_qwen = f"data:image;base64,{encoded_image_text}"
    messages = [
        {
            "role": "user",
            "content": [
                {"type": "image", "image": base64_qwen},
                {"type": "text", "text": "What does the image show?"},
            ],
        }
    ]
    text = processor.apply_chat_template(
        messages, tokenize=False, add_generation_prompt=True
    )
    image_inputs, video_inputs = process_vision_info(messages)

    # tokenize
    return processor(
        text=[text],
        images=image_inputs,
        videos=video_inputs,
        padding=False,
        max_length=MAX_SEQUENCE_LENGTH,
        truncation=True,
    )


ds = ds.map(preprocess_and_tokenize, remove_columns=ds.column_names)


# Define a oneshot data collator for multimodal inputs.
def data_collator(batch):
    assert len(batch) == 1
    return {key: torch.tensor(value) for key, value in batch[0].items()}


# Recipe
recipe = [
    GPTQModifier(
        targets="Linear",
        scheme="W4A16",
        ignore=["lm_head", "re:visual.*", "re:model.visual.*"],
    ),
]

# Perform oneshot
oneshot(
    model=model,
    tokenizer=model_id,
    dataset=ds,
    recipe=recipe,
    max_seq_length=MAX_SEQUENCE_LENGTH,
    num_calibration_samples=NUM_CALIBRATION_SAMPLES,
    trust_remote_code_model=True,
    data_collator=data_collator,
    sequential_targets=["Qwen2_5_VLDecoderLayer"],
)

# Confirm generations of the quantized model look sane.
print("========== SAMPLE GENERATION ==============")
dispatch_for_generation(model)
messages = [
    {
        "role": "user",
        "content": [
            {
                "type": "image",
                "image": "http://images.cocodataset.org/train2017/000000231895.jpg",
            },
            {"type": "text", "text": "Please describe the animal in this image\n"},
        ],
    }
]
prompt = processor.apply_chat_template(messages, add_generation_prompt=True)
image_inputs, video_inputs = process_vision_info(messages)
inputs = processor(
    text=[prompt],
    images=image_inputs,
    videos=video_inputs,
    padding=False,
    max_length=MAX_SEQUENCE_LENGTH,
    truncation=True,
    return_tensors="pt",
).to("cuda")
output = model.generate(**inputs, max_new_tokens=100)
print(processor.decode(output[0], skip_special_tokens=True))
print("==========================================")


# Save to disk compressed.
SAVE_DIR = f"/data/llm/{model_id.split('/')[-1]}-W4A16-4096-v0.6.0-20250708"
model.save_pretrained(SAVE_DIR, save_compressed=True)
processor.save_pretrained(SAVE_DIR)
print(f"Model saved to {SAVE_DIR}")
```
</details>

**Comment by KarlDe1** [comment]
Hi @kylesayrs 

Another question: when quantizing, what is **the minimum block size** that I can set? What is **the minimum block size supported by vLLM**?

From my testing, it seems that setting block_size to 32 slightly improves accuracy.

I’d like to try improving accuracy this way.

**Comment by kylesayrs** [comment]
Hi @KarlDe1 

> Thank you for your reply. Is it possible to recover some of the accuracy through certain methods? For example, by ignoring more layers during quantization?

Yes, although you will see less performance benefits. On a related note, support for recipes with multiple types of precision (FP4, FP8, ect) are also in progress.

> so perhaps the pattern should be updated to "re:model.visual.*"?

The script provided on HF might be out of date, as the [HF checkpoint has changed](https://github.com/huggingface/transformers/pull/37033/files) since that model was quantized. The latest script in the examples folder should have the correct ignore list.

> I tested the ChartQA (test, CoT) dataset using a quantized model - Qwen2.5-VL-3B-Instruct-W4A16, and obtained an anywhere_in_answer_relaxed_correctness score of 71.96, which is lower than the 78.96 reported in RedHatAI's paper.

I suspect that using an out-of-date ignore list may mean that your quantized model has quantized the vision tower. This can lead to significant accuracy loss for very little performance benefit, which is why the vision tower is typically ignored.

> Another question: when quantizing, what is the minimum block size that I can set? What is the minimum block size supported by vLLM?

Can you give more context about this "block_size" argument? It's not an argument I'm familiar with, are you perhaps referring to group size?

**Comment by KarlDe1** [comment]
Hi @kylesayrs,

Thank you for the prompt reply.

> Can you give more context about this "block_size" argument? It's not an argument I'm familiar with, are you perhaps referring to group size?

Yes, block_size is the same as group_size.

**Comment by kylesayrs** [comment]
@KarlDe1 

> Another question: when quantizing, what is the minimum block size that I can set? What is the minimum block size supported by vLLM?
> From my testing, it seems that setting block_size to 32 slightly improves accuracy.

There's a good discussion [here](https://github.com/vllm-project/llm-compressor/issues/1525) the about tradeoffs of smaller group sizes. TLDR, group sizes smaller than the standard 128 can lead to better accuracy and more tensor parallelism support, but will have increased latency and are not guaranteed to have a supporting kernel.

**Comment by kylesayrs** [comment]
Good discussion here! Feel free to open another ticket or reach out of the vLLM slack if there's anything else you'd like to discuss!
